src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php

```diff
@@ -349,6 +352,26 @@ public function viewAction(int $orderId, Request $request): Response
         ]);
     }
 
+    /**
+     * @param OrderForViewing $orderForViewing
+     */
+    private function handleOutOfStockProduct(OrderForViewing $orderForViewing)
+    {
+        $isStockManagementEnabled = $this->configuration->get('PS_STOCK_MANAGEMENT');
+        if (!$isStockManagementEnabled || $orderForViewing->isDelivered() || $orderForViewing->isShipped()) {
+            return;
+        }
+
+        foreach ($orderForViewing->getProducts()->getProducts() as $product) {
+            if ($product->getAvailableQuantity() <= 0) {
+                $this->addFlash(
+                    'warning',
+                    $this->trans('This product is out of stock: ', 'Admin.Orderscustomers.Notification') . ' ' . $product->getName()
```

Comment:
Does this mean we add a flash notif for each of out-of-stock product ?

Reply:
It is actually the case :)

Reply:
So if the BO admin checks an old order for which all products are out of stock for a long time, he gets drown in notifs 😅?

Reply:
We don't change the behavior :)